scripts/src/main/resources/scripts/command/yarn

```diff
@@ -31,7 +31,7 @@ function doSetup() {
     then
       doEcho "Version ${installed_version} of yarn is already installed at $(command -v yarn)"
     fi
-  elif [ "${1}" != "silent" ] || [ "${version}" != "latest" ]
+  elif [ "${version}" != "latest" ] || [ -z "${installed_version}" ]
```

Comment:
Why do we have this `[ "${version}" != "latest" ]` here?
If `[ -z "${installed_version}" ]` makes perfectly sense as then there is no yarn installed so we have to install it.
However, if `${version}"` is `latest` then we technically do not know if the installation is up-to-date or not.
IMHO the intention of the old `if` condition was to omit this re-installation of `latest` version in `silent` mode but still have it on an explicit call of `devon yarn setup . This does not work anymore after this change so IMHO it is still incorrect.

So should we also add `|| [ "${1}" != "silent" ]`?
I think then we would have all cases covered or am I missing something?

Reply:
I simply edited this so this is resolved.